docs/docs/schema/pipeline.json

```diff
@@ -31,7 +31,7 @@
             "description": "Input topic",
             "properties": {
                 "role": {
-                    "description": "Custom identifier belonging to a topic, provide only if `type` is `extra` or `extra-pattern`",
+                    "description": "Custom identifier belonging to a topic, provide only if `type` is `extra`, `pattern` or `extra-pattern`",
```

Comment:
There is no need for type extra and extra-pattern anymore

Reply:
I updated the way it works

Reply:
i would prefer to keep the old functionality at least until v2 is released

Reply:
WDYT?

Reply:
I get that this is shorter (saves 1 line each), but I am not a fan personally. Not only does it look a bit odd without the value, it would also mean that we have to change our data model to an optional: `TopicConfig | None` unless there's a way to mitigate it with a Pydantic root validator. Either way, it adds complexity which is unnecessary imo for such small gain

Reply:
Fixed I think